cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -532,13 +532,13 @@ struct BooleanAnyImpl : public ScalarAggregator {
     }
     if (batch[0].is_scalar()) {
       const Scalar& scalar = *batch[0].scalar;
-      this->has_nulls = !scalar.is_valid;
-      this->any = scalar.is_valid && checked_cast<const BooleanScalar&>(scalar).value;
-      this->count += scalar.is_valid;
+      this->has_nulls |= !scalar.is_valid;
+      this->any |= scalar.is_valid && checked_cast<const BooleanScalar&>(scalar).value;
```

Comment:
Why the `|`? Isn't that necessary only in `MergeFrom`?

Reply:
See https://github.com/apache/arrow/pull/43799#issuecomment-2308439586